modules/services/MapRouletteService.js

```diff
@@ -0,0 +1,384 @@
+import RBush from 'rbush';
+
+import { AbstractSystem } from '../core/AbstractSystem';
+import { Task as MapRouletteTask } from '../maproulette/Task';
+import { utilFetchResponse } from '../util';
+import { marked } from 'marked';
+
+const TILEZOOM = 14;
+const MAPROULETTE_API = 'https://maproulette.org/api/v2';
+
+
+/**
+ * `MapRouletteService`
+
+ * Events available:
+ *   'loadedData'
+ */
+export class MapRouletteService extends AbstractSystem {
+
+  /**
+   * @constructor
+   * @param  `context`  Global shared application context
+   */
+  constructor(context) {
+    super(context);
+    this.id = 'maproulette';
+    this.autoStart = false;
+
+    this._cache = null;   // cache gets replaced on init/reset
+    this._deletedChallenges = {};
+    this._tiler = new Tiler().zoomRange(TILEZOOM).skipNullIsland(true);
+  }
+
+
+  /**
+   * initAsync
+   * Called after all core objects have been constructed.
+   * @return {Promise} Promise resolved when this component has completed initialization
+   */
+  initAsync() {
+    return this.resetAsync();
+  }
+
+
+  /**
+   * startAsync
+   * Called after all core objects have been initialized.
+   * @return {Promise} Promise resolved when this component has completed startup
+   */
+  startAsync() {
+      this._started = true;
+  }
+
+
+  /**
+   * resetAsync
+   * Called after completing an edit session to reset any internal state
+   * @return {Promise} Promise resolved when this component has completed resetting
+   */
+  resetAsync() {
+    if (this._cache) {
+      Object.values(this._cache.inflightTile).forEach(controller => this._abortRequest(controller));
+    }
+    this._cache = {
+      tasks: new Map(),    // Map (taskID -> Task)
+      loadedTile: {},
+      inflightTile: {},
+      inflightPost: {},
+      closed: {},
+      comment: {},
+      rtree: new RBush()
+    };
+    this._deletedChallenges = {};
+    return Promise.resolve();
+  }
+
+
+  /**
+   * getData
+   * Get already loaded data that appears in the current map view
+   * @return  {Array}  Array of data
+   */
+  getData() {
+    const extent = this.context.systems.map.extent();
+    let challengeId = d3_select('.challenge-id-input').property('value');
+    challengeId = challengeId ? Number(challengeId) : null;
+    return this._cache.rtree.search(extent.bbox())
+      .map(d => d.data)
+      .filter(task => !challengeId || task.task.parentId === challengeId);
+  }
+
+
+  /**
+   * loadTiles
+   * Schedule any data requests needed to cover the current map view
+   */
+  loadTiles() {
+    // determine the needed tiles to cover the view
+    const viewport = this.context.viewport;
+    const tiles = this._tiler.getTiles(viewport).tiles;
+
+    // abort inflight requests that are no longer needed
+    this._abortUnwantedRequests(this._cache, tiles);
+
+    // get the challenge ID entered by the user
+    let challengeId = Number(d3_select('.challenge-id-input').property('value'));
```

Comment:
same code smell as before - `MapRouletteService` should own the challengeID, not be d3-selecting it out of a field that is managed elsewhere.